flake8_trio.py

```diff
@@ -202,28 +204,21 @@ def has_decorator(decorator_list: list[ast.expr], *names: str):
 
 # matches the fully qualified name against fnmatch pattern
 # used to match decorators and methods to user-supplied patterns
-def fnmatch_qualified_name(name_list: list[ast.expr], *patterns: str):
-    def construct_name(expr: ast.expr) -> str | None:
-        if isinstance(expr, ast.Call):
-            expr = expr.func
-        if isinstance(expr, ast.Name):
-            return expr.id
-        elif isinstance(expr, ast.Attribute):
-            attr = construct_name(expr.value)
-            assert attr is not None
-            return attr + "." + expr.attr
-        # See https://peps.python.org/pep-0614/ - we don't handle everything
-        return None  # pragma: no cover  # impossible on Python 3.8
-
+def fnmatch_qualified_name(name_list: list[ast.expr], *patterns: str) -> str | None:
+    if sys.version_info[:2] < (3, 9):
+        return None  # pragma: no cover
```

Comment:
Given that this is causing us coverage issues, affects 107/108 too, and I'd expect it to be trouble w.r.t. unused-ignores config across versions, let's just drop support for Python 3.8 entirely.  Drop it from the CI matrix, update the requires_python metadata, and pass `shed --py39-plus` in our tox.ini.